sstables/compaction_manager.cc

```diff
@@ -639,7 +660,9 @@ future<> compaction_manager::rewrite_sstables(column_family* cf, sstables::compa
             reevaluate_postponed_compactions();
             return make_ready_future<stop_iteration>(stop_iteration::yes);
         });
-    }).finally([this, task] {
+        });
+    }).finally([this, task, wrk = std::move(wrk)] {
+        _stats.pending_tasks -= wrk->total_pending;
```

Comment:
Shouldn't this be 0 at this point?

Reply:
if we ran all tasks yes, if it was stopped in the middle then no.

This is the finally block